Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1839,7 +1839,7 @@ void MediaPlayerPrivateGStreamer::handleMessage(GstMessage* message)
 
         if (GST_CLOCK_TIME_IS_VALID(gstreamerPosition))
             playbackPosition = MediaTime(gstreamerPosition, GST_SECOND);
-        if (!eosFlagIsSetInSink && playbackPosition.isValid() && duration.isValid()
+        if (m_isLegacyPlaybin && !eosFlagIsSetInSink && playbackPosition.isValid() && duration.isValid()
```

Comment:
So the issue doesn't happen with playbin2?
Maybe this should be done for all non-looping cases... `!m_player->isLooping()`